README.md

```diff
@@ -1,2 +1,56 @@
+
+  This project already contains the finished models that it needs, however you first need to create the migrations required to create the database in the first place. For that, go to your console and write `python -m manage.py makemigrations`. This will create the migrations required for a database to be created and for the models to map to said database. By default, the database created is an SQLite3 database.
+
+- **Run the migrations**
+  
+  Finally, to apply the migration changes, go to your console and run the command `python -m manage.py migrate`. This will create a database for your data to be stored.
+
+  **YOUR DATABASE MAY CONTAIN SENSITIVE INFORMATION, SO NEVER SHARE IT WITH ANYONE YOU DON'T TRUST**
+
+- **Enable stripe**
+
+  Payments are processed using Stripe. You will need to make an account on Stripe first by going [here](https://dashboard.stripe.com/register), in order to get your own API keys. Once your account is created, you will need to activate it, and then go to your dashboard. In the middle of the page, click on "API keys for developers".
+  
+  ![εικόνα](https://github.com/ArceusLegend/Ecommerce/assets/109414442/1a767a99-9156-40cc-a921-aa99aab07ca9)
+
+  Now go to the project, and navigate to core/settings.py Scroll all the way to the bottom to find the `PUBLISHABLE_KEY`, `SECRET_KEY`, and `STRIPE_ENDPOINT_SECRET` constants. Replace these with your own keys; the publishable and secret keys can be found in the API keys tab:
+  
+  ![εικόνα](https://github.com/ArceusLegend/Ecommerce/assets/109414442/e08eb6f2-4c42-432d-ab05-9fa63084b2a6)
+
+  You can then find the endpoint secret by going to the "Webhooks" tab, and then clicking on either button you want:
+
+  ![εικόνα](https://github.com/ArceusLegend/Ecommerce/assets/109414442/938a13d3-c4c4-424f-846e-38f129f9bde5)
+
+  Both buttons will take you to the same page, which will include a sample code block on the right side of your screen. Go to line 25, and copy the `endpoint_secret` key. This is your STRIPE_ENDPOINT_SECRET.
+  
+  Finally, activate stripe with `stripe listen --forward-to localhost:8000/payment/webhook/`.
+
+  Note: the port number 8000 is the default port that Django uses to run the server. If you're using a different port number, you should replace 8000 with that instead. You can test that it works by going to the payment page and using the test card numbers in stripe.txt. *No auth* means stripe will simulate a successful payment without the need to authenticate the payment, *Auth* means Stripe will generate a pop up simulating an event that requires authentication to proceed, and *Error* means that stripe will simulate a failed authentication event.
+
+# Run the server
+Your setup is done! You can now activate the server by going to the console and using the command `python -m manage.py runserver`. If you want to add products to the store, you can go to 127.0.0.1:8000/admin, and manage your database from there. 
```

Comment:
`runserver` and SQLite is fine for development, but never use this for production. The setup should support running with something like gunicorn. please see https://docs.djangoproject.com/en/4.2/howto/deployment/ 
 and https://fly.io/django-beats/deploying-django-to-production/ 

I'd suggest to create a parametric `settings.py` which reads envvars and decides which database engine to use. 

I'm attaching an example such settings file from an old project

```
import os
import sys
from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
from django.core.exceptions import ImproperlyConfigured

BASE_DIR = Path(__file__).resolve().parent.parent

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/

# Application definition
DEBUG_ENABLE = os.getenv("DEBUG_ENABLE", "false").lower() == "true"
# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = os.getenv("DEBUG", "false").lower() == "true"

TESTING = "pytest" in sys.modules or "test" in sys.argv

ALLOWED_HOSTS = ["*"]

INSTALLED_APPS = [
    "django.contrib.admin",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.messages",
    "django.contrib.staticfiles",
    "django.contrib.humanize",
    # external apps #
    "corsheaders",
    "healthz",
    "rest_framework",
    "rest_framework_serializer_extensions",
    "rest_framework.authtoken",
    "notifications",
    "fontawesome_free",
    # project apps #
    "users",
    "streams",
    "retro",
    "payments",
]


MIDDLEWARE = [
    "healthz.middleware.HealthCheckMiddleware",
    "corsheaders.middleware.CorsMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.security.SecurityMiddleware",
    "django.contrib.sessions.middleware.SessionMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.csrf.CsrfViewMiddleware",
    "django.contrib.auth.middleware.AuthenticationMiddleware",
    "django.contrib.messages.middleware.MessageMiddleware",
    "django.middleware.clickjacking.XFrameOptionsMiddleware",
]

# for healthz
HEALTHCHECK = {"READINESS_CHECKS": ("databases",)}

# production deployment on Gcloud needs these
CORS_ALLOWED_ORIGINS = [
    "https://storage.googleapis.com",
    "https://invahub.com",
]
# CORS_ALLOW_ALL_ORIGINS = True
CSRF_TRUSTED_ORIGINS = ["https://invahub.com"]

ROOT_URLCONF = "invahub.urls"

TEMPLATES = [
    {
        "BACKEND": "django.template.backends.django.DjangoTemplates",
        "APP_DIRS": True,
        "OPTIONS": {
            "context_processors": [
                "django.template.context_processors.debug",
                "django.template.context_processors.request",
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",
            ],
        },
    },
]

WSGI_APPLICATION = "invahub.wsgi.application"

AUTH_USER_MODEL = "users.InvahubUser"

LOGIN_URL = "/user-auth/login"
LOGIN_REDIRECT_URL = "/streams/dashboard"
LOGOUT_REDIRECT_URL = "/"

# Database
# https://docs.djangoproject.com/en/4.0/ref/settings/#databases

DATABASE_ENGINE = os.getenv("DATABASE_ENGINE", "postgres").lower()  # supports 'sqlite3' or 'postgres' or mysql.

# SETUP DATABASES
if DATABASE_ENGINE == "sqlite3":
    DATABASE_LOCATION = os.getenv("DB_LOCATION", "db.sqlite3")  # always relative to project dir
    DATABASES = {
        "default": {
            "ENGINE": "django.db.backends.sqlite3",
            "NAME": BASE_DIR / DATABASE_LOCATION,
        }
    }
elif DATABASE_ENGINE == "postgres":
    from django.db.backends import postgresql

    DB_SERVER = os.getenv("DB_SERVER", "127.0.0.1")
    DB_PORT = os.getenv("DB_PORT", "5432")
    DB_NAME = os.getenv("DB_NAME", "invahub")
    DB_USER = os.getenv("DB_USERNAME", "devel")
    DB_PASS = os.getenv("DB_PASSWORD", "1q2w3e4r")
    DATABASES = {
        "default": {
            "ENGINE": "django.db.backends.postgresql_psycopg2",
            "HOST": DB_SERVER,
            "PORT": DB_PORT,
            "NAME": DB_NAME,
            "USER": DB_USER,
            "PASSWORD": DB_PASS,
        }
    }
elif DATABASE_ENGINE == "mysql":
    DB_SERVER = os.getenv("DB_SERVER", "127.0.0.1")
    DB_PORT = os.getenv("DB_PORT", "3306")
    DB_NAME = os.getenv("DB_NAME", "invahub")
    DB_USER = os.getenv("DB_USERNAME", "devel")
    DB_PASS = os.getenv("DB_PASSWORD", "1q2w3e4r")
    MYSQL_CONNECTION_MAX_AGE = int(os.getenv("MYSQL_CONNECTION_MAX_AGE", 600))
    DATABASES = {
        "default": {
            "ENGINE": "django.db.backends.mysql",
            "HOST": DB_SERVER,
            "PORT": DB_PORT,
            "NAME": DB_NAME,
            "USER": DB_USER,
            "PASSWORD": DB_PASS,
            "CONN_MAX_AGE": MYSQL_CONNECTION_MAX_AGE,
        }
    }
else:
    raise ImproperlyConfigured("Unknown database engine '{!s}'".format(DATABASE_ENGINE))

# OVERRIDE databases with SQLite3 when testing
if TESTING:
    DATABASE_LOCATION = "db_test.sqlite3"
    DATABASES = {
        "default": {
            "ENGINE": "django.db.backends.sqlite3",
            "NAME": BASE_DIR / DATABASE_LOCATION,
        }
    }

# Password validation
# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.CommonPasswordValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
    },
]

# logging
DATABASE_LOGGING = os.getenv("DATABASE_LOGGING", "false").lower() == "true"
ALL_DJANGO_LOGGING = os.getenv("ALL_DJANGO_LOGGING", "false").lower() == "true"
DEFAULT_LOG_LEVEL = os.getenv("DEFAULT_LOG_LEVEL", None)
if DEFAULT_LOG_LEVEL is None:
    if DEBUG:
        DEFAULT_LOG_LEVEL = "DEBUG"
    else:
        DEFAULT_LOG_LEVEL = "INFO"
if TESTING:
    DEFAULT_LOG_LEVEL = "DEBUG"
    DATABASE_LOGGING = True

LOG_FORMATTERS = {
    "verbose": {
        "format": "{levelname} {asctime} {module} {process:d} {thread:d} {message}",
        "style": "{",
    },
    "simple": {
        "format": "{levelname} ({asctime}): {message}",
        "style": "{",
    },
}

LOG_HANDLERS = {
    "console": {
        "level": DEFAULT_LOG_LEVEL,
        "class": "logging.StreamHandler",
        "formatter": "simple",
    },
    "mail_admins": {
        "level": "ERROR",
        "filters": ["require_debug_false"],
        "class": "django.utils.log.AdminEmailHandler",
    },
}

LOGGERS = {
    "django.request": {
        "handlers": ["console"],
        "level": DEFAULT_LOG_LEVEL,
        "propagate": False,
    },
    "django.command": {
        "handlers": ["console"],
        "level": DEFAULT_LOG_LEVEL,
        "propagate": False,
    },
}
if ALL_DJANGO_LOGGING is True:
    LOGGERS.update(
        {
            "django": {
                "handlers": ["console"],
                "level": DEFAULT_LOG_LEVEL,
                "propagate": False,
            }
        }
    )
if DATABASE_LOGGING is True:
    LOGGERS.update(
        {
            "django.db.backends": {
                "handlers": ["console"],
                "level": "DEBUG",
                "propagate": False,
            }
        }
    )

# Full Logging config
LOGGING = {
    "version": 1,
    "disable_existing_loggers": False,
    "filters": {"require_debug_false": {"()": "django.utils.log.RequireDebugFalse"}},
    "formatters": LOG_FORMATTERS,
    "handlers": LOG_HANDLERS,
    "loggers": LOGGERS,
}


REST_FRAMEWORK = {
    "DEFAULT_PERMISSION_CLASSES": ("rest_framework.permissions.AllowAny",),
    "DEFAULT_AUTHENTICATION_CLASSES": [
        "rest_framework.authentication.SessionAuthentication",
    ],
    "DEFAULT_PARSER_CLASSES": ("rest_framework.parsers.JSONParser",),
    "DEFAULT_RENDERER_CLASSES": (
        "rest_framework.renderers.JSONRenderer",
        "rest_framework.renderers.BrowsableAPIRenderer",
    ),
    "DEFAULT_THROTTLE_CLASSES": [
        "rest_framework.throttling.AnonRateThrottle",
        "rest_framework.throttling.UserRateThrottle",
    ],
    "DEFAULT_THROTTLE_RATES": {"anon": "100/hour", "user": "100/hour"},
    "COERCE_DECIMAL_TO_STRING": False,
    "PAGE_SIZE": 20,
    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
    "DEFAULT_FILTER_BACKENDS": ("django_filters.rest_framework.DjangoFilterBackend",),
    "DEFAULT_SCHEMA_CLASS": "invahub.schema.OpenAPISchema",
}

# Internationalization
# https://docs.djangoproject.com/en/4.0/topics/i18n/

LANGUAGE_CODE = "en-us"

TIME_ZONE = "UTC"

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.0/howto/static-files/

STATIC_URL = os.getenv("STATIC_URL", "static/")
STATIC_ROOT = os.path.join(BASE_DIR, "static")

MEDIA_URL = os.getenv("MEDIA_URL", "media/")
MEDIA_ROOT = os.path.join(BASE_DIR, "media")

# Default primary key field type
# https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field
DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"

```